lib/offline/storage.js

```diff
@@ -524,9 +524,14 @@ shaka.offline.Storage = class {
           pendingContent.size = size;
           this.config_.offline.progressCallback(pendingContent, progress);
         },
-        (initData) => {
+        (initData, systemId) => {
           if (needsInitData && this.config_.offline.usePersistentLicense) {
-            drmEngine.newInitData('cenc', initData);
+            const drmInfo = drmEngine.getDrmInfo();
+            const currentSystemId =
+                shaka.offline.Storage.defaultSystemIds_.get(drmInfo.keySystem);
```

Comment:
The current key system doesn't change after initialization.  So you should compute this above so you only do it once.

Reply:
Ok, I change it. Thanks!